client/pool.c

```diff
@@ -592,22 +592,19 @@ static void calculate_nopaid_shares(struct connection_pool_data *conn_data, stru
 		} else if(diff > conn_data->maxdiff[i]) {
 			conn_data->maxdiff[i] = diff;
 		}
-		// Adding share for miner
-		if(conn_data->miner) {
-			if(conn_data->miner->task_time < task_time) {
-				conn_data->miner->task_time = task_time;
 
-				if(conn_data->miner->maxdiff[i] > 0) {
-					conn_data->miner->prev_diff += conn_data->miner->maxdiff[i];
-					conn_data->miner->prev_diff_count++;
-				}
+		// Adding share for miner
+		if(conn_data->miner->task_time < task_time) {
```

Comment:
conn_data->miner could be null when the calculate_nopaid_share is called before pool received the second request from miner.